### DISCUSSION TITLE
Retrieve list of names/variables for an expression

### DISCUSSION THREAD

**Discussion by geoffwhittington** [danthedeckie/simpleeval#109] [Q&A]
Is it possible to get the list of names used in an expression? I'm looking for something like `variables()` from https://github.com/Axiacore/py-expression-eval

I've seen https://stackoverflow.com/questions/33554036/how-to-get-all-variable-and-method-names-used-in-script but I have a feeling I need these names defined ahead of time as per the example.

Any type of guidance would be helpful!

**Comment by danthedeckie** [comment]
You could wrap the name lookup function with one that stores all names?

You'd also probably want to compile the expression without actually evaluating it?

I like the idea of being able to do this easily without hacking said around too much.
